test/e2e/clusterctl_upgrade_test.go

```diff
@@ -95,6 +99,17 @@ var _ = Describe("When testing clusterctl upgrades (v0.3=>v1.5=>current)", Flake
 				},
 				{ // Upgrade to latest v1beta1.
 					Contract: clusterv1.GroupVersion.Version,
+					PostUpgrade: func(proxy framework.ClusterProxy, namespace, clusterName string) {
+						framework.ValidateCRDMigration(ctx, proxy, namespace, clusterName,
+							func(crd apiextensionsv1.CustomResourceDefinition) bool {
+								return strings.HasSuffix(crd.Name, ".cluster.x-k8s.io") &&
+									crd.Name != "providers.clusterctl.cluster.x-k8s.io" &&
```

Comment:
Not sure what we want to do with providers. At the moment it doesn't matter because it's v1alpha3 and we don't touch it at all

Reply:
maybe make it a overwritable func?

But also good enough to fixup once we start using it in a provider e2e

Reply:
This whole thing is already overridable / configurable if a infra provider runs this test :)

What I meant with providers is the providers CRD. Currently I. did not configure any CRDMigrator to migrate the providers CRD (literally the CRD with the name: `providers.clusterctl.cluster.x-k8s.io`)

Reply:
Ahh, got it.

They only exist when using clusterctl, also clusterctl updates them, so I'd say it should be clusterctl's job to handle them IMHO.

I think we could ignore them as we only have a single api version there, no hooks etc. and currently no plans for a future api version.